onnxruntime/core/providers/cuda/cuda_graph.cc

```diff
@@ -23,16 +22,16 @@ void CUDAGraph::CaptureBegin() {
               "This cuda graph has already captured a graph. "
               "Create a new instance to capture a new graph.");
 
-  CUDA_CALL_THROW(cudaDeviceSynchronize());
-  CUDA_CALL_THROW(cudaStreamBeginCapture(capture_stream_, cudaStreamCaptureModeGlobal));
+  CUDA_CALL_THROW(cudaStreamSynchronize(stream_));
+  CUDA_CALL_THROW(cudaStreamBeginCapture(stream_, cudaStreamCaptureModeThreadLocal));
```

Comment:
Can you please explain this change:  cudaStreamCaptureModeGlobal -> cudaStreamCaptureModeThreadLocal ? And leaving a comment about it would be nice as well

Reply:
Basically `cudaStreamCaptureModeThreadLocal` will allow multiple threads to capture multiple streams separately. More details can be found below:
1. `cudaStreamCaptureModeGlobal`: If the local thread has an ongoing capture sequence that was not initiated with `cudaStreamCaptureModeRelaxed` at `cuStreamBeginCapture`, or if any other thread has a concurrent capture sequence initiated with `cudaStreamCaptureModeGlobal`, this thread is prohibited from potentially unsafe API calls.
2. `cudaStreamCaptureModeThreadLocal`: If the local thread has an ongoing capture sequence not initiated with `cudaStreamCaptureModeRelaxed`, it is prohibited from potentially unsafe API calls. Concurrent capture sequences in other threads are ignored.

I will add a comment in the code as well.